pytorch/pom.xml

```diff
@@ -135,4 +137,45 @@
     </plugins>
   </build>
 
+  <!-- cross-compilation profile for macosx-arm64 builds on macosx-x86_64 CI machines -->
+  <profiles>
+    <profile>
+      <id>pytorch-cross-compile</id>
+      <activation>
+        <property>
+          <name>javacpp.platform</name>
+          <value>macosx-arm64</value>
+        </property>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.bytedeco</groupId>
+            <artifactId>javacpp</artifactId>
+            <dependencies>
+              <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>cpython</artifactId>
+                <version>3.12.1-${project.parent.version}</version>
+                <classifier>${os.name}-${os.arch}</classifier>
+              </dependency>
+              <!-- we need to exclude target cpython until we have a macosx-arm64 build -->
+              <dependency>
+                <groupId>org.bytedeco</groupId>
+                <artifactId>numpy-platform</artifactId>
+                <version>1.26.3-${project.parent.version}</version>
```

Comment:
I suppose we need to do something similar in platform/pom.xml:
```
2024-01-24T00:31:14.1394220Z [ERROR] Failed to execute goal on project pytorch-platform: Could not resolve dependencies for project org.bytedeco:pytorch-platform:jar:2.1.2-1.5.10-SNAPSHOT: Could not find artifact org.bytedeco:cpython:jar:macosx-arm64:3.12.1-1.5.10-SNAPSHOT in sonatype-nexus-snapshots (https://oss.sonatype.org/content/repositories/snapshots) -> [Help 1]
2024-01-24T00:31:14.1403910Z org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project pytorch-platform: Could not resolve dependencies for project org.bytedeco:pytorch-platform:jar:2.1.2-1.5.10-SNAPSHOT: Could not find artifact org.bytedeco:cpython:jar:macosx-arm64:3.12.1-1.5.10-SNAPSHOT in sonatype-nexus-snapshots (https://oss.sonatype.org/content/repositories/snapshots)
```

Reply:
Actually, I don't think we need to include those optional dependencies at all...